python/private/render_pkg_aliases.bzl

```diff
@@ -0,0 +1,187 @@
+        name = name,
+        selects = "\n{}    ".format(
+            "".join([
+                "        {}: {},\n".format(repr(k), repr(v))
+                for k, v in selects.items()
+            ]),
+        ),
+    )
+
+def _render_entrypoints(repo_name, dep):
+    return """\
+load("@{repo_name}_{dep}//:entrypoints.bzl", "entrypoints")
+
+[
+    alias(
+        name = script,
+        actual = "@{repo_name}_{dep}//:" + target,
+        visibility = ["//visibility:public"],
+    )
+    for script, target in entrypoints.items()
+]
+""".format(
+        repo_name = repo_name,
+        dep = dep,
+    )
+
+def _render_common_aliases(repo_name, name, versions = None, default_version = None, rules_python = None):
+    return "\n\n".join([
+        """package(default_visibility = ["//visibility:public"])""",
+        _render_alias(
+            name = name,
+            repo_name = repo_name,
+            dep = name,
+            target = "pkg",
+            versions = versions,
+            default_version = default_version,
+            rules_python = rules_python,
+        ),
+    ] + [
+        _render_alias(
+            name = target,
+            repo_name = repo_name,
+            dep = name,
+            target = target,
+            versions = versions,
+            default_version = default_version,
+            rules_python = rules_python,
+        )
+        for target in ["pkg", "whl", "data", "dist_info"]
+    ])
+
+def render_pkg_aliases(*, repo_name, bzl_packages = None, whl_map = None, rules_python = None, default_version = None):
+    """Create alias declarations for each python dependency.
+
+    The aliases should be appended to the pip_repository BUILD.bazel file. These aliases
+    allow users to use requirement() without needed a corresponding `use_repo()` for each dep
+    when using bzlmod.
+
+    Args:
+        repo_name: the repository name of the parent that is visible to the users.
+        bzl_packages: the list of packages to setup, if not specified, whl_map.keys() will be used instead.
+        whl_map: the whl_map for generating Python version aware aliases.
+        default_version: the default version to be used for the aliases.
+        rules_python: the name of the rules_python workspace.
+
+    Returns:
+        A dict of file paths and their contents.
+    """
+    if not bzl_packages and whl_map:
+        bzl_packages = list(whl_map.keys())
+
+    contents = {}
+    for name in bzl_packages:
+        versions = None
+        if whl_map != None:
+            versions = whl_map[name]
+        name = sanitize_name(name)
+
+        filename = "{}/BUILD.bazel".format(name)
+        contents[filename] = _render_common_aliases(
+            repo_name = repo_name,
+            name = name,
+            versions = versions,
+            rules_python = rules_python,
+            default_version = default_version,
+        )
+
+        if versions == None:
+            filename = "{}/bin/BUILD.bazel".format(name)
```

Comment:
I have added a comment and removed the code for generating aliases if `versions == None`. It would normally happen when using this function from the non-bzlmod setup where the we are not using version aware toolchains. Thanks for raising this point.